services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/MultipartS3AsyncClient.java

```diff
@@ -62,11 +64,13 @@ private MultipartS3AsyncClient(S3AsyncClient delegate, MultipartConfiguration mu
 
     @Override
     public CompletableFuture<PutObjectResponse> putObject(PutObjectRequest putObjectRequest, AsyncRequestBody requestBody) {
+        putObjectRequest = putObjectRequest.toBuilder().checksumAlgorithm(ChecksumAlgorithm.CRC32).build();
         return mpuHelper.uploadObject(putObjectRequest, requestBody);
     }
 
     @Override
     public CompletableFuture<CopyObjectResponse> copyObject(CopyObjectRequest copyObjectRequest) {
+        copyObjectRequest = copyObjectRequest.toBuilder().checksumAlgorithm(ChecksumAlgorithm.CRC32).build();
```

Comment:
Hmm, do we apply checksum for copy operations in crt s3 client?

Reply:
good catch, it seems we don't. removing